### ISSUE TITLE
Power consumption values are different to Multimatic App

### ISSUE THREAD

**Issue by martinkutter** [thomasgermain/vaillant-component#157]
If I use the "VWF 88/4 CENTRAL_HEATING CONSUMED_ELECTRICAL_POWER" sensor as an individual device on the energy dashboard it's comfortable to show power consumption for each day. But if I compare these values with the multimatic app they don't match. For example for yesterday the app shows 7kWh, the dashboard 5kWh. It's different for all days i compared. Do you have any ideas?

**Comment by thomasgermain** [comment]
Hello,

I don't have any idea right now, but let's check that. I don't have any consumption sensors myself, so this is quite hard or me  to test.

Consumption sensors are refreshed every 30min. So you might have different values if, for example:
- consumption for day 1 (between 00:00 to 23:59) is 5kWh
- sensors are refreshed at 00:05 (I mean here, just after 00:00). So this consumption is considered day 2 for HA. So might see some difference here.
- So day 1 will be 5kWh and day 2 xxkWh + 2kWh (that are actually from day 1).

Can't you check, for example, for one week in both HA and multimatic to ensure values are closed to each other ?

I can also release a new version which will fetch consumption sensors more often (but it will only reduce the error margin, there will still be some).


Thomas

**Comment by martinkutter** [comment]
Hey, thank you for your fast answer.
To check this I take some screenshots from the sensor graph in HA and the mutlimatic App.

October 21-23
<img width="750" alt="image" src="https://user-images.githubusercontent.com/16589094/197530626-c4ae7644-899c-4a8e-8852-a201a8bc343b.png">

<img width="250" alt="image" src="https://user-images.githubusercontent.com/16589094/197530402-a849d6b4-5a5a-410a-82bb-6ab65dd3ea77.png"><img width="250" alt="image" src="https://user-images.githubusercontent.com/16589094/197530330-81bfadd5-f931-4ada-9665-645962b1fcd4.png"><img width="250" alt="image" src="https://user-images.githubusercontent.com/16589094/197530240-c08867ab-789c-49a9-a28d-c764b0e5e0c3.png">

As you can see the sensor is only fetched once a day around 12 am. So you are right, the value is half from the past day and half from the current.

**Comment by thomasgermain** [comment]
So, last version https://github.com/thomasgermain/vaillant-component/releases/tag/1.12.9 includes a higher refresh rates.

Maybe values should be closer to what can be seen in the app.

Please let me know how it looks !

**Comment by martinkutter** [comment]
<img width="1452" alt="image" src="https://user-images.githubusercontent.com/16589094/201538325-c20eeaab-b382-4684-97db-1054b2e0d6be.png">

I tested the last version for some days, but unfortunately it behaves the same. Data is only fetched once a day around midday. (Only one day is a little of because of a local network issue.

Anything else I can do?

**Comment by hvbargen** [comment]
Is this a possible duplicate of #135?

**Comment by thomasgermain** [comment]
I released the  1.14.0b1 (beta) version. 

As suggested in another issue, this issue might be related to the server cache.
I added some headers, so the server should not use its cache.
There are two notes:
- I don't even know if the difference between the integration and multimatic is due to the cache
- I can't test it since my installation is not compatible

**Comment by martinkutter** [comment]
Installed. I will come back in a fee days with test results. Mercie!

**Comment by martinkutter** [comment]
<img width="1461" alt="image" src="https://user-images.githubusercontent.com/16589094/218021735-144ba393-5fb9-4593-a668-89ccd3935216.png">

Bad news. Thanks for trying to solve it via the new headers. Unfortunately, the value is still updated only once a day. Please let me know if I can help with testing or troubleshooting in any way.

**Comment by hvbargen** [comment]
I am running this, too. I see several changes per day, which roughly seem to correspond with the heating phases for zone 1.
![Screenshot_2023-02-10-21-48-34-606_io homeassistant companion android](https://user-images.githubusercontent.com/37015738/218194745-04b6ebbd-90f7-4cf4-af8a-843b78762fa3.jpg)

**Comment by ThomasGrRS** [comment]
Good Morning!
I think I have a similar problem.
In the app yesterday I had a largely constant solar production (VRC 700/4 CENTRAL_HEATING SOLAR_YIELD). In the energy dashboard some of it is missing. Instead, there is a solar yield today that was definitely not physically present (totally cloudy).
Maybe this information will help somehow. If I can test something, I will be happy to do so.
Many greetings
Thomas

![Screenshot_20230329_085306_VRC 700](https://user-images.githubusercontent.com/112770585/228452053-837075f0-5365-433c-9f28-6b2189be4df0.jpg)
![Screenshot_20230329_085330_Home Assistant](https://user-images.githubusercontent.com/112770585/228452057-fa9c5606-015d-48c8-88f5-44a8d0171517.jpg)
![Screenshot_20230329_085302_VRC 700](https://user-images.githubusercontent.com/112770585/228452062-1c42b0f6-f2be-468c-9038-6a48fcf5b5c6.jpg)
![Screenshot_20230329_085322_Home Assistant](https://user-images.githubusercontent.com/112770585/228452065-1b7378a8-e521-4945-bb98-26f906c390fe.jpg)

**Comment by thomasgermain** [comment]
@ThomasGrRS are you running the latest version (1.14.0b03) ?

**Comment by ThomasGrRS** [comment]
Ok, now it works fine!
Thanks a lot
